### PR TITLE
[CORL-130] Handle non-visible comment status

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- When a comment is posted, replied to, edited, and the status returned is non-visible, show a message to the user that it is in review.

## How do I test this PR?

- Post/Edit/Write reply to a comment that gets rejected or withheld like posting a comment with a banned word

### CODE REVIEW

File: src/core/client/stream/mutations/CreateCommentMutation.ts [hunk 1]
```diff
@@ -127,10 +157,18 @@ function commit(
...
+      // Skip if comments are always moderated.
+      if (alwaysPremoderated) {
+        return;
+      }
       sharedUpdater(environment, store, input);
       store.get(id)!.setValue(true, "pending");
     },
     updater: store => {
+      // Skip if comments are always moderated.
+      if (alwaysPremoderated) {
```
Comment (wyattjoh) [inline src/core/client/stream/mutations/CreateCommentMutation.ts]: Can you clarify what it means when the update is skipped when premod is enabled?

When premod is enabled for the story, there may be a situation in the future when a comment is automatically accepted/rejected, and I would rather your updater handle this more gracefully.

File: src/core/client/stream/mutations/CreateCommentReplyMutation.ts [hunk 2]
```diff
@@ -10,35 +11,44 @@ import { TalkContext } from "talk-framework/lib/bootstrap";
...
   store: RecordSourceSelectorProxy,
   input: CreateCommentReplyInput
 ) {
+  const commentEdge = store
+    .getRootField("createCommentReply")!
+    .getLinkedRecord("edge")!;
+  const status = commentEdge.getLinkedRecord("node")!.getValue("status");
+
+  // If comment is in review or has been rejected, we don't need to add it.
+  if (isInReview(status) || isRejected(status)) {
```
Comment (wyattjoh) [inline src/core/client/stream/mutations/CreateCommentReplyMutation.ts]: Would it not be better here to rephrase this to say, `isVisable` instead? This way we can have more granular control over when a comment is visible instead of these special cases.

File: src/core/client/stream/mutations/CreateCommentReplyMutation.ts [hunk 3]
```diff
@@ -159,10 +186,18 @@ function commit(
...
+      // Skip if comments are always moderated.
+      if (alwaysPremoderated) {
+        return;
+      }
       sharedUpdater(environment, store, input);
       store.get(id)!.setValue(true, "pending");
     },
     updater: store => {
+      // Skip if comments are always moderated.
+      if (alwaysPremoderated) {
```
Comment (wyattjoh) [inline src/core/client/stream/mutations/CreateCommentReplyMutation.ts]: Same comment as above.

File: src/core/client/stream/mutations/CreateCommentMutation.ts (hunk 1 above)
Comment (alinhle) [reply src/core/client/stream/mutations/CreateCommentMutation.ts]: That makes sense, I'll adapt it!

Comment (alinhle) [reply src/core/client/stream/mutations/CreateCommentMutation.ts]: Good catch @wyattjoh  😄

File: src/core/client/stream/mutations/CreateCommentReplyMutation.ts (hunk 2 above)
Comment (alinhle) [reply src/core/client/stream/mutations/CreateCommentReplyMutation.ts]: Makes sense 😄

**Review by wyattjoh — APPROVED** [review]
